sdk/servicebus/service-bus/src/serviceBusClient.ts

```diff
@@ -233,11 +228,7 @@ export class ServiceBusClient {
    */
   static createFromAadTokenCredentials(
     host: string,
-    credentials:
-      | ApplicationTokenCredentials
-      | UserTokenCredentials
-      | DeviceTokenCredentials
-      | MSITokenCredentials,
+    credentials: any,
```

Comment:
This is user facing api and having types for credentials helps our users to figure out what exactly can they pass in as credentials.

Is this change done because the tests for `createFromAadTokenCredentials` were failing or just building the browser bits and browser tests were failing?
If it is the former, then as mentioned in https://github.com/Azure/azure-sdk-for-js/issues/2163#issuecomment-491348759, we have #2556 to track AAD auth for browser mode and we can have the tests for `createFromAadTokenCredentials` added then

Reply:
It is the latter, browserification fails if the import from ms-rest and it's dependent modules are pulled into browser. Looks like that dependency is not browser friendly.

Reply:
https://github.com/Azure/ms-rest-browserauth is the package we will end up using once we work on #2556 

The credentials from there are of the type [ServiceClientCredentials](https://github.com/Azure/ms-rest-js/blob/master/lib/credentials/serviceClientCredentials.ts)

The 4 kinds of credentials we use today are from https://github.com/Azure/ms-rest-nodeauth. But these credentials extend from the same [ServiceClientCredentials](https://github.com/Azure/ms-rest-js/blob/master/lib/credentials/serviceClientCredentials.ts).

So, maybe we should update `createFromAadTokenCredentials` to be of type `ServiceClientCredentials`. That should work for both node and browser.

Can you try that?

Reply:
On trying and analyzing, found that `.getToken()` API doesn't exist on the ServiceClientCredentials interface and so it doesn't seem true that the current credential types extend from this. 

But also, looks like excluding AAD related tests was sufficien. Reference to `ServiceBusClient.ts` file and it having the ms-rest related import was not problematic as such.

So yes, adding in the tests can be taken up as part of https://github.com/Azure/azure-sdk-for-js/issues/2556. The importing of types seems to work fine.